src/main/java/appeng/client/render/model/MemoryCardBakedModel.java

```diff
@@ -0,0 +1,187 @@
+			AEColor.GRAY, AEColor.GRAY, AEColor.GRAY, AEColor.GRAY,
+	};
+
+	private final VertexFormat format;
+
+	private final IBakedModel baseModel;
+
+	private final TextureAtlasSprite texture;
+
+	private final AEColor[] colorCode;
+
+	private final Cache<AEColor[], MemoryCardBakedModel> modelCache;
+
+	private final ImmutableList<BakedQuad> generalQuads;
+
+	MemoryCardBakedModel( VertexFormat format, IBakedModel baseModel, TextureAtlasSprite texture )
+	{
+		this( format, baseModel, texture, DEFAULT_COLOR_CODE, createCache() );
+	}
+
+	private MemoryCardBakedModel( VertexFormat format, IBakedModel baseModel, TextureAtlasSprite texture, AEColor[] hash, Cache<AEColor[], MemoryCardBakedModel> modelCache )
+	{
+		this.format = format;
+		this.baseModel = baseModel;
+		this.texture = texture;
+		this.colorCode = hash;
+		this.generalQuads = ImmutableList.copyOf( this.buildGeneralQuads() );
+		this.modelCache = modelCache;
+	}
+
+	private static Cache<AEColor[], MemoryCardBakedModel> createCache()
+	{
+		return CacheBuilder.newBuilder()
+				.maximumSize( 100 )
+				.build();
+	}
+
+	@Override
+	public List<BakedQuad> getQuads( @Nullable IBlockState state, @Nullable EnumFacing side, long rand )
+	{
+
+		List<BakedQuad> quads = this.baseModel.getQuads( state, side, rand );
+
+		if( side != null )
+		{
+			return quads;
+		}
+
+		List<BakedQuad> result = new ArrayList<>( quads.size() + this.generalQuads.size() );
+		result.addAll( quads );
+		result.addAll( this.generalQuads );
+		return result;
+	}
+
+	private List<BakedQuad> buildGeneralQuads()
+	{
+		CubeBuilder builder = new CubeBuilder( this.format );
+
+		builder.setTexture( this.texture );
+
+		for( int x = 0; x < 4; x++ )
+		{
+			for( int y = 0; y < 2; y++ )
+			{
+				final AEColor color = this.colorCode[x + y * 4];
+
+				builder.setColorRGB( color.mediumVariant );
+				builder.addCube( 7 + x, 8 + y, 7.5f, 7 + x + 1, 8 + y + 1, 8.5f );
+			}
+		}
+		return builder.getOutput();
+	}
+
+	@Override
+	public boolean isAmbientOcclusion()
+	{
+		return this.baseModel.isAmbientOcclusion();
+	}
+
+	@Override
+	public boolean isGui3d()
+	{
+		return this.baseModel.isGui3d();
+	}
+
+	@Override
+	public boolean isBuiltInRenderer()
+	{
+		return this.baseModel.isBuiltInRenderer();
+	}
+
+	@Override
+	public TextureAtlasSprite getParticleTexture()
+	{
+		return this.baseModel.getParticleTexture();
+	}
+
+	@Override
+	public ItemCameraTransforms getItemCameraTransforms()
+	{
+		return this.baseModel.getItemCameraTransforms();
+	}
+
+	@Override
+	public ItemOverrideList getOverrides()
+	{
+		return new ItemOverrideList( Collections.emptyList() )
+		{
+			@Override
+			public IBakedModel handleItemState( IBakedModel originalModel, ItemStack stack, World world, EntityLivingBase entity )
+			{
+				try
+				{
+					if( stack.getItem() instanceof IMemoryCard )
+					{
+						final IMemoryCard memoryCard = (IMemoryCard) stack.getItem();
+						final AEColor[] colorCode = memoryCard.getColorCode( stack );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "colorCode" which hides the field declared at line 51. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)